src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -1879,6 +1878,10 @@ data CommonTableExpressionKind
 data CommonTableExpressionClause =
     CommonTableExpressionClause CommonTableExpressionKind Ident (IdentInfo -> (TLB.Builder, [PersistValue]))
 
+data ReturningClause
+    = ReturningNothing  -- ^ The default, absent clause.
+    | ReturningStar     -- ^ @RETURNING@ is present.
```

Comment:
If we use `Maybe ReturningClause` as the `SideData` type, then this  has one fewer case:

```suggestion
data ReturningClause
    = ReturningStar     -- ^ @RETURNING@ is present.
```

But `ReturningStar` itself isn't carring any useful information - we could further fold that `Maybe ReturningClause` into just a `Bool`.

Reply:
Yep you're right; I initially tried something different, this had more constructors — but it ended up isomorphic to Maybe ().

Will refactor.